### ISSUE TITLE
🐛 BUG: A request to the Cloudflare API (/memberships) failed.  [code: 9109]

### ISSUE THREAD

**Issue by edmooney** [cloudflare/wrangler2#1422]
### What version of `Wrangler` are you using?

2.0.16

### What operating system are you using?

Linux (Manjaro)

### Describe the Bug

Steps 1-3 work fine per:
https://developers.cloudflare.com/pub-sub/get-started/guide/ - validated the pubsub token - "This API Token is valid and active".
exported the token variable to the console. and verified that the token has all account access with Editing rights.

However get the following message when creating a namespace:
----
    ~  wrangler pubsub namespace create thenamespacename                                                                     ✔ 

✘ [ERROR] A request to the Cloudflare API (/memberships) failed.

  Unauthorized to access requested resource [code: 9109]
  
  If you think this is a bug, please open an issue at:
  https://github.com/cloudflare/wrangler2/issues/new/choose

**Comment by edmooney** [comment]
I had upgraded wrangler, re-authenticated with wrangler login. account id: 9afa0a18632286ba47d3e1da53bb20e5

**Comment by edmooney** [comment]
Generated new key... with Edit rights for pubsub in all accounts - API test curl test with new key:
------------------------------------------------------
    ~  curl -X GET "https://api.cloudflare.com/client/v4/user/tokens/verify" \                                                                    1 ✘
     -H "Authorization: Bearer [btoken-here] \
     -H "Content-Type:application/json"

{"result":{"id":"f408bbc6b902a7c0a9c68b328279b9a5","status":"active"},"success":true,"errors":[],"messages":[{"code":10000,"message":"This API Token is valid and active","type":null}]}%    

---------------------------------------------------------
wrangler pubsub namespace create [namespace-name] 
Retrieving cached values for account from node_modules/.cache/wrangler
Creating Pub/SubNamespace [namespace-name] ...

✘ [ERROR] A request to the Cloudflare API (/accounts/9afa0a18632286ba47d3e1da53bb20e5/pubsub/namespaces) failed.

  Pubsub not enabled [code: 2007]
  
  If you think this is a bug, please open an issue at:
  https://github.com/cloudflare/wrangler2/issues/new/choose

**Comment by kombucha** [comment]
I'm facing the same issue after upgrading from `2.0.14` to `2.0.21`.

```
$ wrangler publish --env=beta
 ⛅️ wrangler 2.0.21 
--------------------

✘ [ERROR] A request to the Cloudflare API (/memberships) failed.

  Invalid request headers [code: 6003]
  - Invalid format for X-Auth-Key header [code: 6103]
```

I tried to re-login but it doesn't help.
Reverting back to 2.0.14 fixes the issue.

**Comment by petebacondarwin** [comment]
@kombucha - in your case it looks like Wrangler is trying to use a global API key that you are providing. The ability to do this was added after 2.0.14. Do you have the `CLOUDFLARE_API_KEY` or `CF_API_KEY` environment variable set? If so, can you check that this is a valid key?

**Comment by kombucha** [comment]
@petebacondarwin indeed I had an invalid `CLOUDFLARE_API_KEY` value in my environment variables from a long time ago. Removing it solved the issue for me.
Thank you!

**Comment by bytemain** [comment]
I'm using https://github.com/cloudflare/wrangler-action to publish worker in GitHub Actions and I meet the same problem.

```
Using API Token authentication
$ Running: wrangler publish --env prod
 ⛅️ wrangler 2.0.21 (update available 2.0.[22])
---------------------------------------------
✘ [ERROR]A request to the Cloudflare API (/memberships) failed.
  Unauthorized to access requested resource [code: 9109]
  If you think this is a bug, please open an issue at: https://github.com/cloudflare/wrangler2/issues/new/choose
```

the key is created by `Edit Cloudflare Workers`:
![CleanShot 2022-07-19 at 13 33 11@2x](https://user-images.githubusercontent.com/13938334/179672642-555cbb93-f71d-4a9a-980c-842aa0151529.png)

**Comment by petebacondarwin** [comment]
@bytemain - when you created this API token from the template, did you restrict the token to only one Zone?

**Comment by threepointone** [comment]
A workaround to this issue is to add an account_id in your wrangler.toml, or set it as an environment variable with CLOUDFLARE_ACCOUNT_ID. We could probably add this tip in the error message itself.

**Comment by marcelduin** [comment]
I also just had this issue (wrangler 2.0.23) with an older project. (Re-)adding the `account_id` to `wrangler.toml` worked.

But it's unclear that this is required, especially because a recent project created with Wrangler 2 doesn't require this.

**Comment by JacobMGEvans** [comment]
Addressed in #1683
